test/lib/jdk/test/lib/net/SimpleHttpServer.java

```diff
@@ -22,140 +23,134 @@
+ **/
 public class SimpleHttpServer {
-    HttpServer _httpserver;
-    ExecutorService _executor;
-
-    String _address;
-
-    String _context, _docroot;
-    int _port;
-
-    public SimpleHttpServer(String context, String docroot) {
+    private final HttpServer httpServer;
+    private ExecutorService executor;
+    private String address;
+    private final String context;
+    private final String docRoot;
+    private final int port;
+    private final InetAddress inetAddress;
+
+    public SimpleHttpServer(final String context, final String docRoot) throws IOException {
         //let the system pick up an ephemeral port in a bind operation
-        this(0, context, docroot);
+        this.port = 0;
+        this.context = context;
+        this.docRoot = docRoot;
+        httpServer = HttpServer.create();
+        //let the server use wild card address
+        inetAddress = null;
     }
 
-    public SimpleHttpServer(int port, String context, String docroot) {
-        _port = port;
-        _context = context;
-        _docroot = docroot;
+    public SimpleHttpServer(final InetAddress inetAddress, final String context, final String docRoot) throws
+            IOException {
+        //let the system pick up an ephemeral port in a bind operation
+        this.port = 0;
+        this.context = context;
+        this.docRoot = docRoot;
+        httpServer = HttpServer.create();
+        this.inetAddress = inetAddress;
     }
 
-    public void start() {
-        MyHttpHandler handler = new MyHttpHandler(_docroot);
-        InetSocketAddress addr = new InetSocketAddress(_port);
-        try {
-            _httpserver = HttpServer.create(addr, 0);
-        } catch (IOException ex) {
-            throw new RuntimeException("cannot create httpserver", ex);
-        }
+    public SimpleHttpServer(final int port, final String context, final String docRoot) throws IOException {
+        this.port = port;
+        this.context = context;
+        this.docRoot = docRoot;
+        httpServer = HttpServer.create();
+        //let the server use wild card address
+        inetAddress = null;
+    }
 
+    public void start() throws IOException {
+        MyHttpHandler handler = new MyHttpHandler(docRoot);
+        InetSocketAddress addr = inetAddress != null ? new InetSocketAddress(inetAddress, port) : new InetSocketAddress(port);
+        httpServer.bind(addr, 0);
         //TestHandler is mapped to /test
-        HttpContext ctx = _httpserver.createContext(_context, handler);
-
-        _executor = Executors.newCachedThreadPool();
-        _httpserver.setExecutor(_executor);
-        _httpserver.start();
-
-        _address = "http://localhost:" + _httpserver.getAddress().getPort();
+        httpServer.createContext(context, handler);
+        executor = Executors.newCachedThreadPool();
+        httpServer.setExecutor(executor);
+        httpServer.start();
+        address = "http://localhost:" + httpServer.getAddress().getPort();
```

Comment:
It might be a better idea to use the URIBuilder here rather than relying on the local machine configuration to resolve "localhost";